phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -265,12 +332,18 @@ public SingleTableColumnResolver(PhoenixConnection connection, NamedTableNode ta
                if (def.getColumnDefName().getFamilyName() != null) {
                    families.add(new PColumnFamilyImpl(PNameFactory.newName(def.getColumnDefName().getFamilyName()),Collections.<PColumn>emptyList()));
                }
-           }
-           Long scn = connection.getSCN();
-           PTable theTable = new PTableImpl(connection.getTenantId(), table.getName().getSchemaName(), table.getName().getTableName(), scn == null ? HConstants.LATEST_TIMESTAMP : scn, families);
+            }
+            Long scn = connection.getSCN();
+            String schema = table.getName().getSchemaName();
+            if (connection.getSchema() != null) {
```

Comment:
The code in discussion will be used during **creation** of **mapped views** only .. It will never be used anywhere else.
So when user provide table name without schema then connection schema should be used if set and if user is providing tablename with schema then we should ignore connection schema. I think this is how most of the databases work.

for eg:- connection schema is set to 'S' (`phoenix> USE 'S'`)
so , if user create mapped view,
`create view A.T(pk ...)`. //then this will map to A.T table only
`create view T(pk..)` // //then this will map to S.T table

And to map table to default schema (User can unset connection schema back to null by using `USE DEFAULT`)
create view T(pk..) // //then this will map to T table

May be I'm missing something, would you mind giving some examples in terms of sql statement.